packages/react-core/src/core/providers/thirdweb-wallet-provider.tsx

```diff
@@ -312,17 +312,39 @@ export function ThirdwebWalletProvider(
         return;
       }
 
-      const personalWalletInfo = walletInfo.connectParams?.personalWallet;
+      let _personalWalletInfo = walletInfo.connectParams?.personalWallet;
+
+      // when connecting to magicLink with social login, it redirects to other page
+      // before redirecting, we save the walletInfo to local storage so that we can auto connect after redirect back to current page
+      // when using smartWallet + magicLink combination - the walletInfo will only contain info about magicLink and not smartWallet because it was never connected
+      // so if smartWallet + magicLink combination is used, we need to connect magicLink first and then connect smartWallet
+      if (
+        walletInfo.walletId === walletIds.magicLink &&
+        walletInfo.connectParams &&
+        "oauthProvider" in walletInfo.connectParams
```

Comment:
@MananTank I don't think we should add react specific code in react-core, can this be fixed from react?